docs/RFCS/20190731_cluster_name.md

```diff
@@ -0,0 +1,110 @@
+- Feature Name: cluster name
+- Status: draft
+- Start Date: 2019-07-31
+- Authors: knz, ben, marc
+- RFC PR: (PR # after acceptance of initial draft)
+- Cockroach Issue: #16784 #15888
+
+# Summary
+
+New feature: a string value, called "cluster name", and checked for
+equality when a newly started node joins a cluster.
+
+This will prevent newly added nodes to join the wrong cluster when a
+user has multiple clusters running side by side.
+
+This would be configured using another command-line parameter `--join-name`
+alongside `--join`.
+
+It will increase operational simplicity and the overall goal to “make
+data easy”
+
+Other impact:
+
+- (should have) it would be checked for equality in the node
```

Comment:
I would love this part because it would make https://github.com/cockroachdb/cockroach/issues/28408 easier. There is still a question of where this goes in the certificate and whether the field is configurable (eg: one of the many fields of the  [Subject](https://tools.ietf.org/html/rfc5280#section-4.1.2.4)). The details of the certificate implementation are probably out of scope for this.